### PR TITLE
HPCC-15035 Document the parallel language changes

### PR DESCRIPTION
**Pull request by JamesDeFabia**
Signed-off-by: Jim DeFabia jamesdefabia@lexisnexis.com

@RichardTaylorHPCC Thanks for your help. 

@ghalliday Please review

### CODE REVIEW

File: docs/ECLLanguageReference/ECLR_mods/BltInFunc-AGGREGATE.xml [hunk 1]
```diff
@@ -93,6 +93,51 @@
...
           </row>
+          <row>
+            <entry><emphasis role="bold">UNORDERED</emphasis></entry>
+
+            <entry>Optional. Specifies the output record order is not significant.</entry>
+          </row>
+          <row>
+            <entry><emphasis role="bold">ORDERED</emphasis></entry>
+
+            <entry>Specifies the significance of the output record order.</entry>
```
Comment (ghalliday) [inline docs/ECLLanguageReference/ECLR_mods/BltInFunc-AGGREGATE.xml]: How about "Specifies whether the output order is significant."

File: docs/ECLLanguageReference/ECLR_mods/BltInFunc-AGGREGATE.xml [hunk 2]
```diff
@@ -93,6 +93,51 @@
...
+          </row>
+          <row>
+            <entry><emphasis>bool</emphasis></entry>
+
+            <entry>When False, specifies the output record order is not significant. When True, specifies the default output record order.</entry>
```
Comment (ghalliday) [inline docs/ECLLanguageReference/ECLR_mods/BltInFunc-AGGREGATE.xml]: I would have expected this to be included in the ordered entry, not as a separate one.

File: docs/ECLLanguageReference/ECLR_mods/BltInFunc-AGGREGATE.xml [hunk 3]
```diff
@@ -93,6 +93,51 @@
...
+          </row>
+          <row>
+            <entry><emphasis role="bold">PARALLEL</emphasis></entry>
+
+            <entry>Optional. Try to evaluate this activity in parallel.</entry>
+          </row>
+          <row>
+            <entry><emphasis>numthreads</emphasis></entry>
+
+            <entry>Optional. Try to evaluate this activity using <emphasis>numthreads</emphasis> threads.</entry>
```
Comment (ghalliday) [inline docs/ECLLanguageReference/ECLR_mods/BltInFunc-AGGREGATE.xml]: This should really be combined with the previous entry.

File: docs/ECLLanguageReference/ECLR_mods/BltInFunc-AGGREGATE.xml [hunk 4]
```diff
@@ -93,6 +93,51 @@
...
+          </row>
+          <row>
+            <entry><emphasis role="bold">ALGORITHM</emphasis></entry>
+
+            <entry>Optional. Override the algorithm used for this activity.</entry>
+          </row>
+          <row>
+            <entry><emphasis>name</emphasis></entry>
+
+            <entry>The algorithm to use for this activity. Must be from the list of supported algorithms for the SORT function's STABLE and UNSTABLE options.</entry>
```
Comment (ghalliday) [inline docs/ECLLanguageReference/ECLR_mods/BltInFunc-AGGREGATE.xml]: This should be combined with the previous entry.  The algorithm depends on which activity this is applied to - it should not talk about sort.

**Comment by ghalliday** [comment]
A general comment:  I really do not like the duplication.  It makes it very hard to maintain - e.g., any changes from this code review now need to be done in lots of places.  Also, the relevance of the options depends on the activity.  By highlighting the options on each activity it is going to pose the question what algorithms are supported for filter, aggregate, join, x, y, z, and the answer will be none, and what is the guaranteed ordering for each operation - which we don't define anywhere at the moment. (We should.)

Can you structure the document so there is an entry for common options - and a link to that from each of the separate activities.  It should include all those above, HINT, and possibly LOCAL.

Even if that isn't possible it would be better to arrange the source to avoid the duplication.
@richardkchapman  do you agree/disagree?

@JamesDeFabia

**Comment by JamesDeFabia** [comment]
@RichardTaylorHPCC can you comment?

**Pull request merged**